include/SFML/Window/WindowHandle.hpp

```diff
@@ -44,8 +44,13 @@ namespace sf
 
 #elif defined(SFML_SYSTEM_LINUX) || defined(SFML_SYSTEM_FREEBSD) || defined(SFML_SYSTEM_OPENBSD) || defined(SFML_SYSTEM_NETBSD)
 
+#if defined(SFML_WAYLAND)
+    // Window handle is struct wl_display* (void*) on Unix - Wayland
+    typedef void* WindowHandle;
+#else
     // Window handle is Window (unsigned long) on Unix - X11
     typedef unsigned long WindowHandle;
+#endif
```

Comment:
We recently migrated to C++17, so we'd prefer using `using` (pun not intended) :) 

```suggestion
#if defined(SFML_WAYLAND)
    // Window handle is struct wl_display* (void*) on Unix - Wayland
    using WindowHandle = void*;
#else
    // Window handle is Window (unsigned long) on Unix - X11
    using WindowHandle = unsigned long;
#endif
```